prober/http.go

```diff
@@ -243,6 +330,38 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 	if resp == nil {
 		resp = &http.Response{}
 	}
+	var (
+		earlierstCertExpiry time.Time
+		i                   = 0
+	)
+	for _, trace := range tt.traces {
+		// We get the duration for the first request from chooseProtocol
+		if i != 0 {
+			durationGaugeVec.WithLabelValues("resolve").Add(trace.dnsDone.Sub(trace.start).Seconds())
+		}
+		if resp.TLS != nil {
+			exp := getEarliestCertExpiry(resp.TLS)
+			if earlierstCertExpiry.IsZero() || earlierstCertExpiry.Before(exp) {
+				earlierstCertExpiry = exp
+			}
+
+			probeSSLEarliestCertExpiryGauge.Set(float64(getEarliestCertExpiry(resp.TLS).UnixNano() / 1e9))
+			if httpConfig.FailIfSSL {
+				success = false
+			}
+			durationGaugeVec.WithLabelValues("connect").Add(trace.connectDone.Sub(trace.dnsDone).Seconds())
+			durationGaugeVec.WithLabelValues("tls").Add(trace.gotConn.Sub(trace.dnsDone).Seconds())
```

Comment:
This is going to be inconsistent with the name of the equivalent metric for the tcp probe

Reply:
Yes.. As I said, I feel like the http duration metric is with these labels much more convinient to use.
We could also make the tcp probe and all other probes put these timings into one duration gauge for consistency. What do you think about that? That would be a breaking change though.

Reply:
I'm not sure what the best way forward is here, http having multiple requests messes things up a bit. Http phases don't also apply to other probes.

Reply:
> I'm not sure what the best way forward is here, http having multiple requests messes things up a bit

I'm already summing up the times for the redirects, so there is only one timeseries per phase.


> Http phases don't also apply to other probes.

Every probe has some sort of phases, right? They are even similar: resolve/connect/tls/processing/transfer should apply 1:1 to tcp probe. A subset of that would work for udp too. For ICMP we could still have resolve/waiting or something like that. DNS is tricky though given that Exchange() is pretty high level. But that could still use the same gauge with a single label.

In the end I'm fine with having 'inconsistent' metrics or just expose times like tls and dns twice.

Reply:
> resolve/connect/tls/processing/transfer should apply 1:1 to tcp probe

Not quite, HTTP is a single query and response whereas TCP can have multiple queries and responses in what HTTP would consider "transfer".

Reply:
Oh right now I see what you mean, didn't use the tcp probes much.
There is probably no way to get something consistent here. I'm leaning towards just exposing the tls and dns timings twice.

Reply:
That's probably the best route, we'll just need to be clear in the HELP strings about how they're different.